data_scrapers/san_francisco_county.py

```diff
@@ -4,101 +4,238 @@
 from typing import Dict
 
 # API endpoints 
-age_gender_url = 'https://data.sfgov.org/resource/sunc-2t3k.json'
-race_ethnicity_url = 'https://data.sfgov.org/resource/vqqm-nsqg.json'
-transmission_url = 'https://data.sfgov.org/resource/tvq9-ec9w.json'
-hospitalizations_url = 'https://data.sfgov.org/resource/nxjg-bhem.json'
-tests_url = 'https://data.sfgov.org/resource/nfpa-mg4g.json'
-
-def get_json(url, query=''):
+# landing page: https://data.sfgov.org/stories/s/San-Francisco-COVID-19-Data-and-Reports/fjki-2fab
+
+RESOURCE_IDS = {'cases_deaths_transmission': 'tvq9-ec9w', 'age_gender': 'sunc-2t3k', 
+                'race_eth': 'vqqm-nsqg', 'tests': 'nfpa-mg4g' }
+
+metadata_url = 'https://data.sfgov.org/api/views/metadata/v1/'
+data_url = 'https://data.sfgov.org/resource/'
+age_gender_url = f"{data_url}{RESOURCE_IDS['age_gender']}.json"
+race_ethnicity_url = f"{data_url}{RESOURCE_IDS['race_eth']}.json"
+transmission_url = f"{data_url}{RESOURCE_IDS['cases_deaths_transmission']}.json"
+tests_url = f"{data_url}{RESOURCE_IDS['tests']}.json"
```

Comment:
This isn’t a huge deal, but you could probably reduce some boilerplate *and* make your code a bit more readable by wrapping this up into a class for dealing with the API, e.g:

```py
class SanFranciscoApi:
    base_url = 'https://data.sfgov.org/'

    def __init__(self):
        # Having a session gets you connection pooling and other
        # performance niceties.
        self.session = requests.Session()
        self.resource_url = urljoin(self.base_url, '/resource/')
        self.metadata_url = urljoin(self.base_url, '/api/views/metadata/v1/')

    def request(self, url, **kwargs):
        response = self.session.request(url, **kwargs)
        response.raise_for_status()
        return response.json()

    def get_cases(self, params):
        return self.request(f'{self.resource_url}tvq9-ec9w', params)

    def get_age_and_gender(self, params):
        return self.request(f'{self.resource_url}sunc-2t3k', params)

    # etc.

# Now in your functions you can have a slightly more readable call
# and not repeat all the business around checking for exceptions and
# parson JSON:
cases_series = api_instance.get_cases({'case_disposition': 'Confirmed',
                                       '$select':'date,sum(case_count) as cases',
                                       '$group':'date',
                                       '$order':'date'})
```

And this gives you more places to do useful other things, like setting a common user-agent string for all your requests.

If any other counties use Socrata, you can split this in two so parts of it are also reusable:

```py
class SocrataApi:
    def __init__(self):
        # Having a session gets you connection pooling and other
        # performance niceties.
        self.session = requests.Session()
        self.resource_url = urljoin(self.base_url, '/resource/')
        self.metadata_url = urljoin(self.base_url, '/api/views/metadata/v1/')

    def request(self, url, **kwargs):
        response = self.session.request(url, **kwargs)
        response.raise_for_status()
        return response.json()

class SanFranciscoApi(SocrataApi):
    base_url = 'https://data.sfgov.org/'

    def get_cases(self, params):
        return self.request(f'{self.resource_url}tvq9-ec9w', params)

    def get_age_and_gender(self, params):
        return self.request(f'{self.resource_url}sunc-2t3k', params)

    # etc.
```